internal/cloud/gcp/compute.go

```diff
@@ -105,6 +108,24 @@ func (g *GCP) ComputeImageImport(bucket, object, imageName, os, region string) (
 
 	// Wait for the build to finish
 	for {
+		select {
+		case <-time.After(30 * time.Second):
+			// Just check the build status below
+		case <-ctx.Done():
+			// cancel the build
+			cancelBuildReq := &cloudbuildpb.CancelBuildRequest{
```

Comment:
If I understand this correctly, in case of `ctx.Done()`, we still return `err == nil` which means that the upload will be marked as a success. I might be missing something though.

Reply:
Great question!

In case it is canceled, an error will be returned. On line no. 139, there is a check of the actual build job status and if it is not `cloudbuildpb.Build_SUCCESS`, an error is returned in all cases. In case the job is canceled, its status will be `cloudbuildpb.Build_CANCELLED`

Reply:
Ah, I see, thanks!